jax/_src/lax/windowed_reductions.py

```diff
@@ -457,8 +457,8 @@ def reduce_window_shape_tuple(operand_shape, window_dimensions, window_strides,
     operand_shape = lax._dilate_shape(operand_shape, base_dilation)
   if window_dilation is not None:
     window_dimensions = lax._dilate_shape(window_dimensions, window_dilation)
-  pads_lo, pads_hi = zip(*padding)
-  operand_padded = core.sum_shapes(operand_shape, pads_lo, pads_hi)
+  pads = zip(*padding)
```

Comment:
My worry here is that now padding of arbitrary shape is accepted. Correct me if I'm wrong, but we only want to accept empty paddings, or paddings of length 2, correct?

Reply:
Actually, it seems like the best fix here is not to make `zip` behave differently, but to replace `zip(*padding)` with `util.unzip2(padding)`, which I think will handle all the corner cases exactly as we need it to.

Reply:
Thanks Jake, this is indeed better, changed!

However, I noticed that having `safe_zip` to work with 0-length inputs is also used in the transpose rule of `reduce_window_sum`:
https://github.com/google/jax/blob/f0112ccda5afe5114855d67845db60fb14907de9/jax/_src/lax/windowed_reductions.py#L365

So I kept the `safe_map` and `safe_zip` changes for now.

I then added the identical test case to the respective autodiff test https://github.com/google/jax/pull/10274/files#diff-22c1f2ce24cf39ea41d2fbba031630102a443aa243435c054e0cbafb6b8183ee

But now I am getting a rather mysterious `fatal Python error`:
https://source.cloud.google.com/results/invocations/768551e1-a1dc-4f3e-9f3f-a35649a3fa0d/targets/jax%2Ftesting%2Finternal_numpy_scipy%2Fpresubmit_github/log
Do you have an idea of what might be causing it?

Reply:
@mattjj do you have ideas how to fix these errors?

The goal is to make calling `lax.reduce_window` and the respective forward/reverse-mode AD on scalars work, e.g.
```python
def f(x):
  return jax.lax.reduce_window(x, 0., jax.lax.add, (), (), ())

f(1.)
grad(f)(1.)
jvp(f (1.,), (1.,))
```

But my suggested fix seems to lead to a `fatal Python error` (https://source.cloud.google.com/results/invocations/605f2ece-dadd-4073-a048-6f366be90a3f/targets/jax%2Ftesting%2Fcpu%2Fpresubmit_github/log) - any suggestions on how to figure out this error, or fix the issue in a different way?